File: code/cpp/test/myswig/classes.cpp

#include <math.h>
#include <iostream>

#include "classes.h"
 
Vector::Vector( double  iX, double iY, double iZ )
	: x(iX), y(iY), z(iZ)
{
	std::cout << "Vector CONSTRUCTOR: " << x << ", " << y << ", " << z << std::endl;
}

Vector::~Vector()
{
	std::cout << "Vector DESTRUCTOR" << std::endl;
}
 
double
Vector::len2() 
{ 
	return x*x + y*y + z*z; 
}

double
Vector::len() 
{ 
	return sqrt(x*x + y*y + z*z); 
}

double
Vector::magnitude() 
{ 
	return sqrt(x*x + y*y + z*z); 
}


Vector getVec( double iX, double iY, double iZ )
{
	return Vector( iX, iY, iZ );
}

Vector* getVecPtr( double iX, double iY, double iZ )
{
	return new Vector ( iX, iY, iZ );
}

File: code/cpp/test/myswig/classes.php
#!/usr/bin/php5 
<?php
require "myphp_module.php";


printf ( "Hello php5\n");

$v = new Vector(2,2,1);
echo $v->len2(), "\n";

// $pV = getVecPtr(4,5,6);
// print $pV->len2();

// $v = getVec(4,5,6);
// print $v->len2();

?>

